### PR TITLE
feat(shorebird_cli): use specific Flutter revision

### PR DESCRIPTION
**Pull request by eseidel**
Move to pulling a specific version of Flutter rather than a branch.

The idea is that `shorebird` is the thing pulled via a branch, and `flutter` is just a versioned dependency.  Lets you know that when you pull a specific version of `shorebird` you get the correct dependencies (where as as it works now you'd end up with some future flutter when pulling an old version of shorebird).

### CODE REVIEW

**Comment by eseidel** [comment]
This depends on https://github.com/shorebirdtech/shorebird/pull/306 landing first, and also needs fixes to the `flutter doctor` commands to not fail.

File: packages/shorebird_cli/test/src/validators/shorebird_flutter_validator_test.dart [hunk 1]
```diff
@@ -14,16 +14,16 @@ class _MockPlatform extends Mock implements Platform {}
 
 void main() {
   group('ShorebirdFlutterValidator', () {
+    const flutterRevision = '45fc514f1a9c347a3af76b02baf980a4d88b7879';
     const gitStatusMessage = """
 On branch stable
```
Comment (eseidel) [inline packages/shorebird_cli/test/src/validators/shorebird_flutter_validator_test.dart]: Did you mean to keep this?

**Review by eseidel — COMMENTED** [review]
lgtm

File: packages/shorebird_cli/test/src/validators/shorebird_flutter_validator_test.dart (hunk 1 above)
Comment (felangel) [reply packages/shorebird_cli/test/src/validators/shorebird_flutter_validator_test.dart]: updated, good catch

**Pull request merged**